libraries/botbuilder-applicationinsights/botbuilder/applicationinsights/flask/flask_telemetry_middleware.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+"""Flask Telemetry Bot Middleware."""
+
+from io import BytesIO
+from threading import current_thread
+
+# Map of thread id => POST body text
+_REQUEST_BODIES = {}
+
+def retrieve_flask_body():
+    """ retrieve_flask_body
+    Retrieve the POST body text from temporary cache.
+    The POST body corresponds with the thread id and should resides in
+    cache just for lifetime of request.
+
+    TODO: Add cleanup job to kill orphans
```

Comment:
Could we perhaps clarify this `TODO`? Is this completed or should we file a separate tracking bug for the future?

Reply:
I removed it comment in both cases.  To hit this, the bots pretty much broken.